// @mantine/core/cjs/components/ColorPicker/ColorPicker.styles.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var createMemoStyles = require('../../theme/create-memo-styles.js');
var getSizeValue = require('../../theme/utils/get-size-value/get-size-value.js');

const sizes = {
  xs: 180,
  sm: 200,
  md: 240,
  lg: 280,
  xl: 320
};
var useStyles = createMemoStyles.createMemoStyles({
  preview: {},
  root: ({ size, fullWidth }) => ({
    boxSizing: "border-box",
    width: fullWidth ? "100%" : getSizeValue.getSizeValue({ size, sizes }),
    padding: 1
  }),
  body: ({ theme, size }) => ({
    display: "flex",
    boxSizing: "border-box",
    paddingTop: getSizeValue.getSizeValue({ size, sizes: theme.spacing }) / 2
  }),
  sliders: ({ theme }) => ({
    flex: 1,
    boxSizing: "border-box",
    "&:not(:only-child)": {
      marginRight: theme.spacing.xs
    }
  }),
  slider: {
    boxSizing: "border-box",
    "& + &": {
      marginTop: 5
    }
  },
  swatch: {
    cursor: "pointer"
  }
});

exports.default = useStyles;
exports.sizes = sizes;
