// @mantine/core/cjs/theme/GlobalStyles.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var reactJss = require('react-jss');
var themeContext = require('./MantineProvider/theme-context.js');
var getFontStyles = require('./utils/get-font-styles/get-font-styles.js');

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const useStyles = reactJss.createUseStyles((theme) => ({
  "@global": {
    "*, *::before, *::after": {
      boxSizing: "border-box"
    },
    body: __spreadProps(__spreadValues({}, getFontStyles.getFontStyles(theme)), {
      backgroundColor: theme.colorScheme === "dark" ? theme.colors.dark[7] : theme.white,
      color: theme.colorScheme === "dark" ? theme.colors.dark[0] : theme.black,
      lineHeight: theme.lineHeight
    })
  }
}), { theming: themeContext.theming });
function GlobalStyles() {
  useStyles();
  return null;
}

exports.GlobalStyles = GlobalStyles;
