// @mantine/core/cjs/components/Progress/Progress.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var cx = require('clsx');
var hooks = require('@mantine/hooks');
var Progress_styles = require('./Progress.styles.js');
var useMantineTheme = require('../../theme/use-mantine-theme/use-mantine-theme.js');
var mergeStyles = require('../../theme/utils/merge-styles/merge-styles.js');
var getThemeColor = require('../../theme/utils/get-theme-color/get-theme-color.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e['default'] : e; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var cx__default = /*#__PURE__*/_interopDefaultLegacy(cx);

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
const PROGRESS_SIZES = Progress_styles.sizes;
function getCumulativeSections(sections) {
  return sections.reduce((acc, section) => {
    acc.sections.push(__spreadProps(__spreadValues({}, section), { accumulated: acc.accumulated }));
    acc.accumulated += section.value;
    return acc;
  }, { accumulated: 0, sections: [] }).sections;
}
function Progress(_a) {
  var _b = _a, {
    className,
    style,
    value,
    color,
    size = "md",
    radius = "sm",
    striped = false,
    themeOverride,
    "aria-label": ariaLabel,
    classNames,
    styles,
    sections
  } = _b, others = __objRest(_b, [
    "className",
    "style",
    "value",
    "color",
    "size",
    "radius",
    "striped",
    "themeOverride",
    "aria-label",
    "classNames",
    "styles",
    "sections"
  ]);
  const theme = useMantineTheme.useMantineTheme(themeOverride);
  const reduceMotion = hooks.useReducedMotion();
  const classes = Progress_styles['default']({ color, size, radius, striped, reduceMotion, theme }, classNames, "progress");
  const _styles = mergeStyles.mergeStyles(classes, styles);
  const segments = Array.isArray(sections) ? getCumulativeSections(sections).map((section, index) => /* @__PURE__ */ React__default.createElement("div", {
    key: index,
    className: classes.bar,
    style: __spreadProps(__spreadValues({}, _styles.bar), {
      width: `${section.value}%`,
      left: `${section.accumulated}%`,
      backgroundColor: getThemeColor.getThemeColor({ theme, color: section.color, shade: 7 })
    })
  })) : null;
  return /* @__PURE__ */ React__default.createElement("div", __spreadValues({
    className: cx__default(classes.root, className),
    style: __spreadValues(__spreadValues({}, style), _styles.root)
  }, others), segments || /* @__PURE__ */ React__default.createElement("div", {
    role: "progressbar",
    "aria-valuemax": 100,
    "aria-valuemin": 0,
    "aria-valuenow": value,
    "aria-label": ariaLabel,
    className: classes.bar,
    style: __spreadProps(__spreadValues({}, _styles.bar), { width: `${value}%` })
  }));
}
Progress.displayName = "@mantine/core/Progress";

exports.PROGRESS_SIZES = PROGRESS_SIZES;
exports.Progress = Progress;
