// @mantine/core/cjs/components/Tooltip/Tooltip.styles.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var createMemoStyles = require('../../theme/create-memo-styles.js');
var getThemeColor = require('../../theme/utils/get-theme-color/get-theme-color.js');
var getFontStyles = require('../../theme/utils/get-font-styles/get-font-styles.js');

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var useStyles = createMemoStyles.createMemoStyles({
  root: {
    position: "relative",
    display: "inline-block"
  },
  wrapper: {
    background: "transparent",
    position: "absolute",
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
    pointerEvents: "none"
  },
  body: ({ theme, color }) => __spreadProps(__spreadValues({}, getFontStyles.getFontStyles(theme)), {
    backgroundColor: getThemeColor.getThemeColor({ theme, color, shade: theme.colorScheme === "dark" ? 3 : 9 }),
    lineHeight: theme.lineHeight,
    fontSize: theme.fontSizes.sm,
    borderRadius: theme.radius.sm,
    padding: [theme.spacing.xs / 2, theme.spacing.xs],
    color: theme.colorScheme === "dark" ? theme.colors.dark[9] : theme.white,
    position: "relative"
  }),
  tooltip: {
    display: "inline-block",
    position: "absolute"
  },
  arrow: ({ theme, color }) => ({
    border: 0,
    background: getThemeColor.getThemeColor({ theme, color, shade: theme.colorScheme === "dark" ? 3 : 9 }),
    zIndex: 1
  })
});

exports.default = useStyles;
