// @mantine/core/cjs/components/Burger/Burger.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var cx = require('clsx');
var hooks = require('@mantine/hooks');
var Burger_styles = require('./Burger.styles.js');
var useMantineTheme = require('../../theme/use-mantine-theme/use-mantine-theme.js');
var mergeStyles = require('../../theme/utils/merge-styles/merge-styles.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e['default'] : e; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var cx__default = /*#__PURE__*/_interopDefaultLegacy(cx);

var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
const BURGER_SIZES = Burger_styles.sizes;
function Burger(_a) {
  var _b = _a, {
    className,
    style,
    opened,
    color,
    size = "md",
    themeOverride,
    elementRef,
    classNames,
    styles
  } = _b, others = __objRest(_b, [
    "className",
    "style",
    "opened",
    "color",
    "size",
    "themeOverride",
    "elementRef",
    "classNames",
    "styles"
  ]);
  const theme = useMantineTheme.useMantineTheme(themeOverride);
  const reduceMotion = hooks.useReducedMotion();
  const _color = color || (theme.colorScheme === "dark" ? theme.white : theme.black);
  const classes = Burger_styles['default']({ color: _color, size, theme, reduceMotion }, classNames, "burger");
  const _styles = mergeStyles.mergeStyles(classes, styles);
  return /* @__PURE__ */ React__default.createElement("button", __spreadValues({
    type: "button",
    className: cx__default(classes.root, className),
    ref: elementRef,
    style: __spreadValues(__spreadValues({}, style), _styles.root)
  }, others), /* @__PURE__ */ React__default.createElement("div", {
    className: cx__default(classes.burger, { [classes.opened]: opened }),
    style: _styles.burger
  }));
}
Burger.displayName = "@mantine/core/Burger";

exports.BURGER_SIZES = BURGER_SIZES;
exports.Burger = Burger;
