// @mantine/core/cjs/components/Code/Code.styles.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var createMemoStyles = require('../../theme/create-memo-styles.js');
var getSharedColorScheme = require('../../theme/utils/get-shared-color-scheme/get-shared-color-scheme.js');
var getFontStyles = require('../../theme/utils/get-font-styles/get-font-styles.js');

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var useStyles = createMemoStyles.createMemoStyles({
  root: ({ theme, color }) => {
    const colors = getSharedColorScheme.getSharedColorScheme({ theme, color, variant: "light" });
    return __spreadProps(__spreadValues({}, getFontStyles.getFontStyles(theme)), {
      lineHeight: theme.lineHeight,
      padding: [1, theme.spacing.xs / 2],
      borderRadius: theme.radius.sm,
      color: theme.colorScheme === "dark" ? theme.white : theme.black,
      backgroundColor: colors.background,
      fontFamily: theme.fontFamilyMonospace,
      fontSize: theme.fontSizes.xs,
      border: `1px solid ${colors.border}`
    });
  },
  block: ({ theme }) => ({
    padding: theme.spacing.xs,
    margin: 0,
    overflowX: "auto"
  })
});

exports.default = useStyles;
