// @mantine/core/esm/components/Tabs/Tabs.styles.js
import { createMemoStyles } from '../../theme/create-memo-styles.js';
import { getSizeValue } from '../../theme/utils/get-size-value/get-size-value.js';

var useStyles = createMemoStyles({
  root: ({ orientation }) => ({
    display: orientation === "vertical" ? "flex" : "block"
  }),
  tabsListWrapper: {},
  tabsList: {},
  pills: ({ orientation }) => ({
    marginRight: orientation === "vertical" ? 20 : 0
  }),
  body: ({ theme, tabPadding, orientation }) => ({
    [orientation === "horizontal" ? "paddingTop" : "paddingLeft"]: getSizeValue({
      size: tabPadding,
      sizes: theme.spacing
    })
  }),
  default: ({ theme, orientation }) => ({
    [orientation === "horizontal" ? "borderBottom" : "borderRight"]: `2px solid ${theme.colorScheme === "dark" ? theme.colors.dark[4] : theme.colors.gray[2]}`,
    "& $tabsList": {
      [orientation === "horizontal" ? "marginBottom" : "marginRight"]: -2
    }
  }),
  outline: ({ theme, orientation }) => ({
    [orientation === "horizontal" ? "borderBottom" : "borderRight"]: `1px solid ${theme.colorScheme === "dark" ? theme.colors.dark[4] : theme.colors.gray[2]}`,
    "& $tabsList": {
      [orientation === "horizontal" ? "marginBottom" : "marginRight"]: -1
    }
  })
});

export default useStyles;
