// @mantine/core/cjs/components/Menu/MenuButton/MenuButton.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var cx = require('clsx');
var MenuButton_styles = require('./MenuButton.styles.js');
var useMantineTheme = require('../../../theme/use-mantine-theme/use-mantine-theme.js');
var mergeStyles = require('../../../theme/utils/merge-styles/merge-styles.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e['default'] : e; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var cx__default = /*#__PURE__*/_interopDefaultLegacy(cx);

var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
function MenuButton(_a) {
  var _b = _a, {
    className,
    style,
    children,
    onHover,
    hovered,
    themeOverride,
    elementRef,
    icon,
    color,
    disabled,
    rightSection,
    component: Element = "button",
    classNames,
    styles,
    radius
  } = _b, others = __objRest(_b, [
    "className",
    "style",
    "children",
    "onHover",
    "hovered",
    "themeOverride",
    "elementRef",
    "icon",
    "color",
    "disabled",
    "rightSection",
    "component",
    "classNames",
    "styles",
    "radius"
  ]);
  const theme = useMantineTheme.useMantineTheme(themeOverride);
  const classes = MenuButton_styles['default']({ color, theme, radius }, classNames, "menu");
  const _styles = mergeStyles.mergeStyles(classes, styles);
  return /* @__PURE__ */ React__default.createElement(Element, __spreadValues({
    type: "button",
    role: "menuitem",
    className: cx__default(classes.item, { [classes.itemHovered]: hovered }, className),
    onMouseEnter: () => !disabled && onHover(),
    ref: elementRef,
    disabled,
    style: __spreadValues(__spreadValues(__spreadValues({}, style), _styles.item), hovered ? _styles.itemHovered : null)
  }, others), /* @__PURE__ */ React__default.createElement("div", {
    className: classes.itemInner,
    style: _styles.itemInner
  }, icon && /* @__PURE__ */ React__default.createElement("div", {
    className: classes.itemIcon,
    style: _styles.itemIcon
  }, icon), /* @__PURE__ */ React__default.createElement("div", {
    className: classes.itemBody,
    style: _styles.itemBody
  }, /* @__PURE__ */ React__default.createElement("div", {
    className: classes.itemLabel,
    style: _styles.itemLabel
  }, children), rightSection)));
}
MenuButton.displayName = "@mantine/core/MenuButton";

exports.MenuButton = MenuButton;
