// @mantine/core/cjs/components/Menu/MenuButton/MenuButton.styles.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var createMemoStyles = require('../../../theme/create-memo-styles.js');
var getSizeValue = require('../../../theme/utils/get-size-value/get-size-value.js');
var getThemeColor = require('../../../theme/utils/get-theme-color/get-theme-color.js');
var getFontStyles = require('../../../theme/utils/get-font-styles/get-font-styles.js');
var hexToRgba = require('../../../theme/utils/hex-to-rgba/hex-to-rgba.js');

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var useStyles = createMemoStyles.createMemoStyles({
  item: ({ theme, color, radius }) => __spreadProps(__spreadValues({}, getFontStyles.getFontStyles(theme)), {
    WebkitTapHighlightColor: "transparent",
    fontSize: theme.fontSizes.sm,
    border: 0,
    backgroundColor: "transparent",
    outline: 0,
    width: "100%",
    textAlign: "left",
    display: "inline-block",
    textDecoration: "none",
    boxSizing: "border-box",
    padding: [theme.spacing.xs, theme.spacing.sm],
    cursor: "pointer",
    borderRadius: getSizeValue.getSizeValue({ size: radius, sizes: theme.radius }),
    color: color ? getThemeColor.getThemeColor({ theme, color, shade: theme.colorScheme === "dark" ? 5 : 7 }) : theme.colorScheme === "dark" ? theme.colors.dark[0] : theme.black,
    "&:disabled": {
      color: theme.colorScheme === "dark" ? theme.colors.dark[3] : theme.colors.gray[5],
      cursor: "not-allowed"
    }
  }),
  itemHovered: ({ theme, color }) => ({
    backgroundColor: color ? hexToRgba.hexToRgba(getThemeColor.getThemeColor({ theme, color, shade: theme.colorScheme === "dark" ? 9 : 0 }), theme.colorScheme === "dark" ? 0.2 : 1) : theme.colorScheme === "dark" ? hexToRgba.hexToRgba(theme.colors.dark[3], 0.35) : theme.colors.gray[0]
  }),
  itemInner: {
    display: "flex",
    alignItems: "center",
    height: "100%"
  },
  itemBody: {
    display: "flex",
    alignItems: "center",
    justifyContent: "space-between",
    flex: 1
  },
  itemIcon: ({ theme }) => ({
    marginRight: theme.spacing.xs,
    "& *": {
      display: "block"
    }
  }),
  itemLabel: {
    lineHeight: 1
  }
});

exports.default = useStyles;
