// @mantine/core/cjs/components/ColorInput/ColorInput.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var hooks = require('@mantine/hooks');
var InputWrapper = require('../InputWrapper/InputWrapper.js');
var Input = require('../Input/Input.js');
var ColorSwatch = require('../ColorSwatch/ColorSwatch.js');
var Popper = require('../Popper/Popper.js');
var Paper = require('../Paper/Paper.js');
var ColorPicker = require('../ColorPicker/ColorPicker.js');
var ColorInput_styles = require('./ColorInput.styles.js');
var useMantineTheme = require('../../theme/use-mantine-theme/use-mantine-theme.js');
var mergeStyles = require('../../theme/utils/merge-styles/merge-styles.js');
var parsers = require('../ColorPicker/converters/parsers.js');
var converters = require('../ColorPicker/converters/converters.js');
var getSizeValue = require('../../theme/utils/get-size-value/get-size-value.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e['default'] : e; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
const SWATCH_SIZES = {
  xs: 16,
  sm: 18,
  md: 22,
  lg: 28,
  xl: 36
};
const ARROW_OFFSET = {
  xs: 12,
  sm: 15,
  md: 17,
  lg: 21,
  xl: 25
};
function ColorInput(_a) {
  var _b = _a, {
    label,
    description,
    error,
    required,
    wrapperProps,
    size = "sm",
    format = "hex",
    onChange,
    onFocus,
    onBlur,
    value,
    defaultValue,
    classNames,
    styles,
    themeOverride,
    disallowInput = false,
    fixOnBlur = true,
    withPreview = true,
    swatchesPerRow = 10,
    withPicker = true,
    icon,
    transition = "pop-top-left",
    id,
    dropdownZIndex = 1,
    transitionDuration = 0,
    transitionTimingFunction,
    className,
    style,
    swatches
  } = _b, others = __objRest(_b, [
    "label",
    "description",
    "error",
    "required",
    "wrapperProps",
    "size",
    "format",
    "onChange",
    "onFocus",
    "onBlur",
    "value",
    "defaultValue",
    "classNames",
    "styles",
    "themeOverride",
    "disallowInput",
    "fixOnBlur",
    "withPreview",
    "swatchesPerRow",
    "withPicker",
    "icon",
    "transition",
    "id",
    "dropdownZIndex",
    "transitionDuration",
    "transitionTimingFunction",
    "className",
    "style",
    "swatches"
  ]);
  const theme = useMantineTheme.useMantineTheme(themeOverride);
  const classes = ColorInput_styles['default']({ theme }, classNames, "color-input");
  const _styles = mergeStyles.mergeStyles(classes, styles);
  const uuid = hooks.useId(id);
  const [referenceElement, setReferenceElement] = React.useState(null);
  const [dropdownOpened, setDropdownOpened] = React.useState(false);
  const [lastValidValue, setLastValidValue] = React.useState("");
  const [_value, setValue] = hooks.useUncontrolled({
    value,
    defaultValue,
    finalValue: "",
    rule: (val) => !!val && val.trim().length > 0,
    onChange
  });
  const handleInputFocus = (event) => {
    typeof onFocus === "function" && onFocus(event);
    setDropdownOpened(true);
  };
  const handleInputBlur = (event) => {
    typeof onBlur === "function" && onBlur(event);
    setDropdownOpened(false);
    fixOnBlur && setValue(lastValidValue);
  };
  React.useEffect(() => {
    if (parsers.isColorValid(_value)) {
      setLastValidValue(_value);
    }
  }, [_value]);
  hooks.useDidUpdate(() => {
    if (parsers.isColorValid(_value)) {
      setValue(converters.convertHsvaTo(format, parsers.parseColor(_value)));
    }
  }, [format]);
  return /* @__PURE__ */ React__default.createElement(InputWrapper.InputWrapper, __spreadValues({
    label,
    description,
    error,
    required,
    classNames,
    styles,
    size,
    id: uuid,
    className,
    style,
    __staticSelector: "color-input"
  }, wrapperProps), /* @__PURE__ */ React__default.createElement("div", {
    ref: setReferenceElement
  }, /* @__PURE__ */ React__default.createElement(Input.Input, __spreadProps(__spreadValues({}, others), {
    __staticSelector: "color-input",
    id: uuid,
    onFocus: handleInputFocus,
    onBlur: handleInputBlur,
    spellCheck: false,
    size,
    value: _value,
    onChange: (event) => setValue(event.currentTarget.value),
    invalid: !!error,
    required,
    icon: icon || (withPreview ? /* @__PURE__ */ React__default.createElement(ColorSwatch.ColorSwatch, {
      color: parsers.isColorValid(_value) ? _value : "#fff",
      size: getSizeValue.getSizeValue({ size, sizes: SWATCH_SIZES })
    }) : null),
    readOnly: disallowInput,
    classNames,
    styles: __spreadProps(__spreadValues({}, styles), {
      input: __spreadProps(__spreadValues({}, styles == null ? void 0 : styles.input), { cursor: disallowInput ? "pointer" : void 0 })
    })
  }))), /* @__PURE__ */ React__default.createElement(Popper.Popper, {
    referenceElement,
    transitionDuration: hooks.useReducedMotion() ? 0 : transitionDuration,
    transitionTimingFunction,
    transition,
    mounted: dropdownOpened,
    position: "bottom",
    placement: "start",
    gutter: 5,
    withArrow: true,
    arrowSize: 3,
    zIndex: dropdownZIndex,
    arrowClassName: classes.arrow,
    arrowStyle: __spreadProps(__spreadValues({}, _styles.arrow), { left: getSizeValue.getSizeValue({ size, sizes: ARROW_OFFSET }) })
  }, /* @__PURE__ */ React__default.createElement("div", {
    style: { pointerEvents: "all" }
  }, /* @__PURE__ */ React__default.createElement(Paper.Paper, {
    shadow: "sm",
    padding: size,
    className: classes.dropdownBody,
    onMouseDown: (event) => event.preventDefault()
  }, /* @__PURE__ */ React__default.createElement(ColorPicker.ColorPicker, {
    __staticSelector: "color-input",
    value: _value,
    onChange: setValue,
    format,
    swatches,
    swatchesPerRow,
    withPicker,
    size,
    focusable: false
  })))));
}
ColorInput.displayName = "@mantine/core/ColorInput";

exports.ColorInput = ColorInput;
