// @mantine/core/esm/components/Group/Group.styles.js
import { createMemoStyles } from '../../theme/create-memo-styles.js';
import { getSizeValue } from '../../theme/utils/get-size-value/get-size-value.js';

const POSITIONS = {
  left: "flex-start",
  center: "center",
  right: "flex-end",
  apart: "space-between"
};
var useStyles = createMemoStyles({
  root: ({
    spacing,
    position,
    noWrap,
    direction,
    theme,
    grow,
    align,
    withGutter
  }) => ({
    display: "flex",
    flexDirection: direction,
    alignItems: align || (direction === "row" ? "center" : grow ? "stretch" : POSITIONS[position]),
    flexWrap: noWrap ? "nowrap" : "wrap",
    justifyContent: direction === "row" && POSITIONS[position],
    margin: -1 * getSizeValue({ size: spacing, sizes: theme.spacing }) / 2,
    padding: withGutter ? getSizeValue({ size: spacing, sizes: theme.spacing }) / 2 : 0
  }),
  child: ({ grow, spacing, theme, count, direction }) => ({
    maxWidth: grow && direction === "row" ? `calc(${100 / count}% - ${getSizeValue({ size: spacing, sizes: theme.spacing })}px)` : void 0,
    flexGrow: grow ? 1 : 0,
    margin: getSizeValue({ size: spacing, sizes: theme.spacing }) / 2
  })
});

export default useStyles;
