// @mantine/core/esm/components/MultiSelect/MultiSelect.js
import React, { useRef, useState } from 'react';
import cx from 'clsx';
import { useId, useUncontrolled, useMergedRef } from '@mantine/hooks';
import { InputWrapper } from '../InputWrapper/InputWrapper.js';
import { Input } from '../Input/Input.js';
import { DefaultValue } from './DefaultValue/DefaultValue.js';
import { DefaultItem } from '../Select/DefaultItem/DefaultItem.js';
import { filterData } from './filter-data/filter-data.js';
import { getSelectRightSectionProps } from '../Select/SelectRightSection/get-select-right-section-props.js';
import { SelectItems } from '../Select/SelectItems/SelectItems.js';
import { SelectDropdown } from '../Select/SelectDropdown/SelectDropdown.js';
import useStyles from './MultiSelect.styles.js';
import { useMantineTheme } from '../../theme/use-mantine-theme/use-mantine-theme.js';
import { mergeStyles } from '../../theme/utils/merge-styles/merge-styles.js';
import { scrollIntoView } from '../../utils/scroll-into-view/scroll-into-view.js';

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
function defaultFilter(value, selected, item) {
  if (selected) {
    return false;
  }
  return item.label.toLowerCase().trim().includes(value.toLowerCase().trim());
}
function MultiSelect(_a) {
  var _b = _a, {
    className,
    style,
    themeOverride,
    required,
    label,
    description,
    size = "sm",
    error,
    classNames,
    styles,
    wrapperProps,
    value,
    defaultValue,
    data,
    onChange,
    valueComponent: Value = DefaultValue,
    itemComponent = DefaultItem,
    id,
    transition = "pop-top-left",
    transitionDuration = 0,
    transitionTimingFunction,
    maxDropdownHeight = 220,
    shadow = "sm",
    nothingFound,
    onFocus,
    onBlur,
    searchable = false,
    placeholder,
    filter = defaultFilter,
    limit = Infinity,
    clearSearchOnChange = true,
    clearable = false,
    clearSearchOnBlur = false,
    clearButtonLabel,
    variant,
    onSearchChange,
    disabled = false,
    initiallyOpened = false,
    radius = "sm",
    elementRef,
    icon
  } = _b, others = __objRest(_b, [
    "className",
    "style",
    "themeOverride",
    "required",
    "label",
    "description",
    "size",
    "error",
    "classNames",
    "styles",
    "wrapperProps",
    "value",
    "defaultValue",
    "data",
    "onChange",
    "valueComponent",
    "itemComponent",
    "id",
    "transition",
    "transitionDuration",
    "transitionTimingFunction",
    "maxDropdownHeight",
    "shadow",
    "nothingFound",
    "onFocus",
    "onBlur",
    "searchable",
    "placeholder",
    "filter",
    "limit",
    "clearSearchOnChange",
    "clearable",
    "clearSearchOnBlur",
    "clearButtonLabel",
    "variant",
    "onSearchChange",
    "disabled",
    "initiallyOpened",
    "radius",
    "elementRef",
    "icon"
  ]);
  const theme = useMantineTheme(themeOverride);
  const classes = useStyles({ theme, size, variant, invalid: !!error }, classNames, "multi-select");
  const _styles = mergeStyles(classes, styles);
  const dropdownRef = useRef();
  const inputRef = useRef();
  const itemsRefs = useRef({});
  const uuid = useId(id);
  const [dropdownOpened, setDropdownOpened] = useState(initiallyOpened);
  const [hovered, setHovered] = useState(-1);
  const [searchValue, setSearchValue] = useState("");
  const handleSearchChange = (val) => {
    typeof onSearchChange === "function" && onSearchChange(val);
    setSearchValue(val);
  };
  const [_value, setValue] = useUncontrolled({
    value,
    defaultValue,
    finalValue: [],
    rule: (val) => Array.isArray(val),
    onChange
  });
  const formattedData = data.map((item) => typeof item === "string" ? { label: item, value: item } : item);
  const handleValueRemove = (_val) => setValue(_value.filter((val) => val !== _val));
  const handleInputChange = (event) => {
    setHovered(0);
    handleSearchChange(event.currentTarget.value);
    setDropdownOpened(true);
  };
  const handleInputFocus = (event) => {
    typeof onFocus === "function" && onFocus(event);
    setDropdownOpened(true);
  };
  const handleInputBlur = (event) => {
    typeof onBlur === "function" && onBlur(event);
    clearSearchOnBlur && handleSearchChange("");
    setDropdownOpened(false);
  };
  const filteredData = filterData({
    data: formattedData,
    searchable,
    searchValue,
    limit,
    filter,
    value: _value
  });
  const handleItemSelect = (item) => {
    clearSearchOnChange && handleSearchChange("");
    if (_value.includes(item.value)) {
      handleValueRemove(item.value);
    } else {
      setValue([..._value, item.value]);
      if (hovered === filteredData.length - 1) {
        setHovered(filteredData.length - 2);
      }
    }
  };
  const handleInputKeydown = (event) => {
    switch (event.nativeEvent.code) {
      case "ArrowUp": {
        event.preventDefault();
        setDropdownOpened(true);
        setHovered((current) => {
          var _a2;
          const nextIndex = current > 0 ? current - 1 : current;
          scrollIntoView(dropdownRef.current, itemsRefs.current[(_a2 = filteredData[nextIndex]) == null ? void 0 : _a2.value]);
          return nextIndex;
        });
        break;
      }
      case "ArrowDown": {
        event.preventDefault();
        setDropdownOpened(true);
        setHovered((current) => {
          var _a2;
          const nextIndex = current < filteredData.length - 1 ? current + 1 : current;
          scrollIntoView(dropdownRef.current, itemsRefs.current[(_a2 = filteredData[nextIndex]) == null ? void 0 : _a2.value]);
          return nextIndex;
        });
        break;
      }
      case "Enter": {
        if (filteredData[hovered]) {
          event.preventDefault();
          handleItemSelect(filteredData[hovered]);
        }
        break;
      }
      case "Backspace": {
        if (_value.length > 0 && searchValue.length === 0 && searchable) {
          setValue(_value.slice(0, -1));
          setDropdownOpened(true);
        }
        break;
      }
      case "Escape": {
        setDropdownOpened(false);
      }
    }
  };
  const selectedItems = _value.map((val) => formattedData.find((item) => item.value === val)).filter((val) => !!val).map((item) => /* @__PURE__ */ React.createElement(Value, __spreadProps(__spreadValues({}, item), {
    disabled,
    className: classes.value,
    style: _styles.value,
    onRemove: () => handleValueRemove(item.value),
    key: item.value,
    themeOverride,
    size,
    styles,
    classNames,
    radius
  })));
  const handleClear = () => {
    var _a2;
    handleSearchChange("");
    setValue([]);
    (_a2 = inputRef.current) == null ? void 0 : _a2.focus();
  };
  const shouldRenderDropdown = filteredData.length > 0 || searchValue.length > 0 && !!nothingFound && filteredData.length === 0;
  return /* @__PURE__ */ React.createElement(InputWrapper, __spreadValues({
    required,
    id: uuid,
    label,
    error,
    description,
    size,
    className,
    style,
    themeOverride,
    classNames,
    styles,
    __staticSelector: "multi-select"
  }, wrapperProps), /* @__PURE__ */ React.createElement("div", {
    className: classes.wrapper,
    style: _styles.wrapper,
    role: "combobox",
    "aria-haspopup": "listbox",
    "aria-owns": `${uuid}-items`,
    "aria-controls": uuid,
    "aria-expanded": dropdownOpened,
    onMouseLeave: () => setHovered(-1),
    tabIndex: -1
  }, /* @__PURE__ */ React.createElement(Input, __spreadValues({
    __staticSelector: "multi-select",
    style: { overflow: "hidden" },
    classNames,
    component: "div",
    multiline: true,
    size,
    variant,
    disabled,
    invalid: !!error,
    required,
    radius,
    icon,
    onMouseDown: (event) => {
      var _a2;
      event.preventDefault();
      (_a2 = inputRef.current) == null ? void 0 : _a2.focus();
    }
  }, getSelectRightSectionProps({
    styles: __spreadProps(__spreadValues({}, styles), {
      input: __spreadProps(__spreadValues({}, styles == null ? void 0 : styles.input), {
        cursor: !searchable ? disabled ? "not-allowed" : "pointer" : void 0
      })
    }),
    size,
    shouldClear: clearable && _value.length > 0,
    themeOverride,
    clearButtonLabel,
    onClear: handleClear,
    error
  })), /* @__PURE__ */ React.createElement("div", {
    className: classes.values,
    style: _styles.values
  }, selectedItems, /* @__PURE__ */ React.createElement("input", __spreadValues({
    ref: useMergedRef(elementRef, inputRef),
    type: "text",
    id: uuid,
    style: _styles.searchInput,
    className: cx(classes.searchInput, {
      [classes.searchInputPointer]: !searchable,
      [classes.searchInputInputHidden]: !dropdownOpened && _value.length > 0 || !searchable && _value.length > 0,
      [classes.searchInputEmpty]: _value.length === 0
    }),
    onKeyDown: handleInputKeydown,
    value: searchValue,
    onChange: handleInputChange,
    onFocus: handleInputFocus,
    onBlur: handleInputBlur,
    readOnly: !searchable,
    placeholder: _value.length === 0 ? placeholder : void 0,
    disabled
  }, others)))), /* @__PURE__ */ React.createElement(SelectDropdown, {
    themeOverride,
    mounted: dropdownOpened && shouldRenderDropdown,
    transition,
    transitionDuration,
    transitionTimingFunction,
    uuid,
    shadow,
    maxDropdownHeight,
    classNames,
    styles,
    elementRef: dropdownRef,
    __staticSelector: "multi-select"
  }, /* @__PURE__ */ React.createElement(SelectItems, {
    data: filteredData,
    hovered,
    themeOverride,
    classNames,
    styles,
    uuid,
    __staticSelector: "multi-select",
    onItemHover: setHovered,
    onItemSelect: handleItemSelect,
    itemsRefs,
    itemComponent,
    size,
    nothingFound
  }))));
}
MultiSelect.displayName = "@mantine/core/MultiSelect";

export { MultiSelect, defaultFilter };
