// @mantine/core/esm/components/Tooltip/Tooltip.js
import React, { useRef, useState } from 'react';
import cx from 'clsx';
import { useReducedMotion } from '@mantine/hooks';
import { Popper } from '../Popper/Popper.js';
import useStyles from './Tooltip.styles.js';
import { useMantineTheme } from '../../theme/use-mantine-theme/use-mantine-theme.js';
import { mergeStyles } from '../../theme/utils/merge-styles/merge-styles.js';

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
function Tooltip(_a) {
  var _b = _a, {
    className,
    style,
    themeOverride,
    label,
    children,
    opened,
    delay = 0,
    gutter = 5,
    color = "gray",
    disabled = false,
    withArrow = false,
    arrowSize = 2,
    position = "top",
    placement = "center",
    transition = "pop-top-left",
    transitionDuration = 100,
    zIndex = 1e3,
    transitionTimingFunction,
    width = "auto",
    wrapLines = false,
    allowPointerEvents = false,
    positionDependencies = [],
    elementRef,
    tooltipRef,
    tooltipId,
    classNames,
    styles
  } = _b, others = __objRest(_b, [
    "className",
    "style",
    "themeOverride",
    "label",
    "children",
    "opened",
    "delay",
    "gutter",
    "color",
    "disabled",
    "withArrow",
    "arrowSize",
    "position",
    "placement",
    "transition",
    "transitionDuration",
    "zIndex",
    "transitionTimingFunction",
    "width",
    "wrapLines",
    "allowPointerEvents",
    "positionDependencies",
    "elementRef",
    "tooltipRef",
    "tooltipId",
    "classNames",
    "styles"
  ]);
  const theme = useMantineTheme(themeOverride);
  const classes = useStyles({ theme, color }, classNames, "tooltip");
  const _styles = mergeStyles(classes, styles);
  const timeoutRef = useRef();
  const [_opened, setOpened] = useState(false);
  const visible = (typeof opened === "boolean" ? opened : _opened) && !disabled;
  const [referenceElement, setReferenceElement] = useState(null);
  const handleOpen = () => {
    window.clearTimeout(timeoutRef.current);
    setOpened(true);
  };
  const handleClose = () => {
    if (delay !== 0) {
      timeoutRef.current = window.setTimeout(() => {
        setOpened(false);
      }, delay);
    } else {
      setOpened(false);
    }
  };
  return /* @__PURE__ */ React.createElement("div", __spreadValues({
    className: cx(classes.root, className),
    ref: elementRef,
    style: __spreadValues(__spreadValues({}, style), _styles.root)
  }, others), /* @__PURE__ */ React.createElement(Popper, {
    referenceElement,
    transitionDuration: useReducedMotion() ? 0 : transitionDuration,
    transition,
    mounted: visible,
    position,
    placement,
    gutter,
    withArrow,
    arrowSize,
    zIndex,
    arrowClassName: classes.arrow,
    arrowStyle: _styles.arrow,
    forceUpdateDependencies: [color, ...positionDependencies]
  }, /* @__PURE__ */ React.createElement("div", {
    className: classes.body,
    ref: tooltipRef,
    style: __spreadProps(__spreadValues({}, _styles.body), {
      pointerEvents: allowPointerEvents ? "all" : "none",
      whiteSpace: wrapLines ? "normal" : "nowrap",
      width
    })
  }, label)), /* @__PURE__ */ React.createElement("div", {
    onMouseEnter: handleOpen,
    onMouseLeave: handleClose,
    onFocusCapture: handleOpen,
    onBlurCapture: handleClose,
    ref: setReferenceElement
  }, children));
}
Tooltip.displayName = "@mantine/core/Tooltip";

export { Tooltip };
