// @mantine/core/cjs/components/MultiSelect/DefaultValue/DefaultValue.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var cx = require('clsx');
var CloseButton = require('../../ActionIcon/CloseButton/CloseButton.js');
var DefaultValue_styles = require('./DefaultValue.styles.js');
var useMantineTheme = require('../../../theme/use-mantine-theme/use-mantine-theme.js');
var mergeStyles = require('../../../theme/utils/merge-styles/merge-styles.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e['default'] : e; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var cx__default = /*#__PURE__*/_interopDefaultLegacy(cx);

var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
const buttonSizes = {
  xs: 16,
  sm: 22,
  md: 24,
  lg: 26,
  xl: 30
};
function DefaultValue(_a) {
  var _b = _a, {
    label,
    classNames,
    styles,
    className,
    style,
    onRemove,
    themeOverride,
    disabled,
    size,
    radius
  } = _b, others = __objRest(_b, [
    "label",
    "classNames",
    "styles",
    "className",
    "style",
    "onRemove",
    "themeOverride",
    "disabled",
    "size",
    "radius"
  ]);
  const theme = useMantineTheme.useMantineTheme(themeOverride);
  const classes = DefaultValue_styles['default']({ theme, size, disabled, radius }, classNames, "multi-select");
  const _styles = mergeStyles.mergeStyles(classes, styles);
  return /* @__PURE__ */ React__default.createElement("div", __spreadValues({
    className: cx__default(classes.defaultValue, className),
    style: __spreadValues(__spreadValues({}, style), _styles.defaultValue)
  }, others), /* @__PURE__ */ React__default.createElement("span", null, label), !disabled && /* @__PURE__ */ React__default.createElement(CloseButton.CloseButton, {
    "aria-hidden": true,
    onClick: onRemove,
    size: buttonSizes[size],
    radius: 2,
    color: "blue",
    variant: "transparent",
    iconSize: buttonSizes[size] / 2,
    className: classes.defaultValueRemove,
    style: _styles.defaultValueRemove,
    tabIndex: -1
  }));
}
DefaultValue.displayName = "@mantine/core/MultiSelect/DefaultValue";

exports.DefaultValue = DefaultValue;
