// @mantine/core/cjs/components/Image/Image.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var cx = require('clsx');
var Text = require('../Text/Text.js');
var ImageIcon = require('./ImageIcon.js');
var Image_styles = require('./Image.styles.js');
var useMantineTheme = require('../../theme/use-mantine-theme/use-mantine-theme.js');
var mergeStyles = require('../../theme/utils/merge-styles/merge-styles.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e['default'] : e; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var cx__default = /*#__PURE__*/_interopDefaultLegacy(cx);

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
function Image(_a) {
  var _b = _a, {
    className,
    style,
    themeOverride,
    alt,
    src,
    fit = "cover",
    width = "100%",
    height = "auto",
    radius = 0,
    imageProps,
    withPlaceholder = false,
    placeholder,
    imageRef,
    elementRef,
    classNames,
    styles,
    caption
  } = _b, others = __objRest(_b, [
    "className",
    "style",
    "themeOverride",
    "alt",
    "src",
    "fit",
    "width",
    "height",
    "radius",
    "imageProps",
    "withPlaceholder",
    "placeholder",
    "imageRef",
    "elementRef",
    "classNames",
    "styles",
    "caption"
  ]);
  const theme = useMantineTheme.useMantineTheme(themeOverride);
  const classes = Image_styles['default']({ radius, theme }, classNames, "image");
  const _styles = mergeStyles.mergeStyles(classes, styles);
  const [loaded, setLoaded] = React.useState(false);
  const [error, setError] = React.useState(!src);
  const isPlaceholder = withPlaceholder && (!loaded || error);
  const firstUpdate = React.useRef(true);
  React.useEffect(() => {
    if (firstUpdate.current) {
      firstUpdate.current = false;
    } else {
      setLoaded(false);
      setError(false);
    }
  }, [src]);
  return /* @__PURE__ */ React__default.createElement("div", __spreadValues({
    className: cx__default(classes.root, className),
    style: __spreadValues(__spreadValues({ width, height }, style), _styles.root),
    ref: elementRef
  }, others), isPlaceholder && /* @__PURE__ */ React__default.createElement("div", {
    className: classes.placeholder,
    title: alt,
    style: _styles.placeholder
  }, placeholder || /* @__PURE__ */ React__default.createElement(ImageIcon.ImageIcon, {
    style: { width: 40, height: 40 }
  })), /* @__PURE__ */ React__default.createElement("figure", {
    className: classes.figure
  }, /* @__PURE__ */ React__default.createElement("img", __spreadValues({
    className: classes.image,
    src,
    alt,
    style: __spreadProps(__spreadValues({}, _styles.image), { objectFit: fit }),
    ref: imageRef,
    onLoad: (event) => {
      setLoaded(true);
      typeof (imageProps == null ? void 0 : imageProps.onLoad) === "function" && imageProps.onLoad(event);
    },
    onError: (event) => {
      setError(true);
      typeof (imageProps == null ? void 0 : imageProps.onError) === "function" && imageProps.onError(event);
    }
  }, imageProps)), !!caption && /* @__PURE__ */ React__default.createElement(Text.Text, {
    component: "figcaption",
    size: "sm",
    align: "center",
    className: classes.caption,
    style: _styles.caption
  }, caption)));
}
Image.displayName = "@mantine/core/Image";

exports.Image = Image;
