// @mantine/core/cjs/components/Tabs/Tabs.styles.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var createMemoStyles = require('../../theme/create-memo-styles.js');
var getSizeValue = require('../../theme/utils/get-size-value/get-size-value.js');

var useStyles = createMemoStyles.createMemoStyles({
  root: ({ orientation }) => ({
    display: orientation === "vertical" ? "flex" : "block"
  }),
  tabsListWrapper: {},
  tabsList: {},
  pills: ({ orientation }) => ({
    marginRight: orientation === "vertical" ? 20 : 0
  }),
  body: ({ theme, tabPadding, orientation }) => ({
    [orientation === "horizontal" ? "paddingTop" : "paddingLeft"]: getSizeValue.getSizeValue({
      size: tabPadding,
      sizes: theme.spacing
    })
  }),
  default: ({ theme, orientation }) => ({
    [orientation === "horizontal" ? "borderBottom" : "borderRight"]: `2px solid ${theme.colorScheme === "dark" ? theme.colors.dark[4] : theme.colors.gray[2]}`,
    "& $tabsList": {
      [orientation === "horizontal" ? "marginBottom" : "marginRight"]: -2
    }
  }),
  outline: ({ theme, orientation }) => ({
    [orientation === "horizontal" ? "borderBottom" : "borderRight"]: `1px solid ${theme.colorScheme === "dark" ? theme.colors.dark[4] : theme.colors.gray[2]}`,
    "& $tabsList": {
      [orientation === "horizontal" ? "marginBottom" : "marginRight"]: -1
    }
  })
});

exports.default = useStyles;
