// @mantine/core/cjs/components/Tabs/Tabs.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var cx = require('clsx');
var hooks = require('@mantine/hooks');
var Group = require('../Group/Group.js');
var Tab = require('./Tab/Tab.js');
var TabControl = require('./TabControl/TabControl.js');
var Tabs_styles = require('./Tabs.styles.js');
var useMantineTheme = require('../../theme/use-mantine-theme/use-mantine-theme.js');
var mergeStyles = require('../../theme/utils/merge-styles/merge-styles.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e['default'] : e; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var cx__default = /*#__PURE__*/_interopDefaultLegacy(cx);

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
function clamp(value, min, max) {
  return Math.min(Math.max(value, min), max);
}
function getPreviousTab(active, tabs) {
  for (let i = active - 1; i >= 0; i -= 1) {
    if (!tabs[i].props.disabled) {
      return i;
    }
  }
  return active;
}
function getNextTab(active, tabs) {
  for (let i = active + 1; i < tabs.length; i += 1) {
    if (!tabs[i].props.disabled) {
      return i;
    }
  }
  return active;
}
function findInitialTab(tabs) {
  for (let i = 0; i < tabs.length; i += 1) {
    if (!tabs[i].props.disabled) {
      return i;
    }
  }
  return -1;
}
function Tabs(_a) {
  var _b = _a, {
    className,
    children,
    style,
    initialTab,
    themeOverride,
    active,
    position = "left",
    grow = false,
    onTabChange,
    color,
    variant = "default",
    classNames,
    styles,
    tabPadding = "xs",
    orientation = "horizontal"
  } = _b, others = __objRest(_b, [
    "className",
    "children",
    "style",
    "initialTab",
    "themeOverride",
    "active",
    "position",
    "grow",
    "onTabChange",
    "color",
    "variant",
    "classNames",
    "styles",
    "tabPadding",
    "orientation"
  ]);
  const theme = useMantineTheme.useMantineTheme(themeOverride);
  const classes = Tabs_styles['default']({ theme, tabPadding, orientation }, classNames, "tabs");
  const _styles = mergeStyles.mergeStyles(classes, styles);
  const controlRefs = React.useRef({});
  const tabs = React__default.Children.toArray(children).filter((item) => item.type === Tab.Tab);
  const [_activeTab, handleActiveTabChange] = hooks.useUncontrolled({
    value: active,
    defaultValue: initialTab,
    finalValue: findInitialTab(tabs),
    rule: (value) => typeof value === "number",
    onChange: onTabChange
  });
  const activeTab = clamp(_activeTab, 0, tabs.length - 1);
  const nextTabCode = orientation === "horizontal" ? "ArrowRight" : "ArrowDown";
  const previousTabCode = orientation === "horizontal" ? "ArrowLeft" : "ArrowUp";
  const handleKeyDown = (event) => {
    if (event.nativeEvent.code === nextTabCode) {
      event.preventDefault();
      const nextTab = getNextTab(activeTab, tabs);
      handleActiveTabChange(nextTab);
      controlRefs.current[nextTab].focus();
    }
    if (event.nativeEvent.code === previousTabCode) {
      event.preventDefault();
      const previousTab = getPreviousTab(activeTab, tabs);
      handleActiveTabChange(previousTab);
      controlRefs.current[previousTab].focus();
    }
  };
  const panes = tabs.map((tab, index) => /* @__PURE__ */ React__default.createElement(TabControl.TabControl, __spreadProps(__spreadValues({}, tab.props), {
    key: index,
    active: activeTab === index,
    tabProps: tab.props,
    onKeyDown: handleKeyDown,
    color,
    variant,
    orientation,
    elementRef: (node) => {
      controlRefs.current[index] = node;
    },
    onClick: () => activeTab !== index && handleActiveTabChange(index),
    classNames,
    styles
  })));
  const content = tabs[activeTab].props.children;
  return /* @__PURE__ */ React__default.createElement("div", __spreadProps(__spreadValues({}, others), {
    className: cx__default(classes.root, className),
    style: __spreadValues(__spreadValues({}, style), _styles.root)
  }), /* @__PURE__ */ React__default.createElement("div", {
    className: cx__default(classes.tabsListWrapper, classes[variant]),
    style: __spreadValues(__spreadValues({}, _styles.tabsListWrapper), _styles[variant])
  }, /* @__PURE__ */ React__default.createElement(Group.Group, {
    className: classes.tabsList,
    style: _styles.tabsList,
    role: "tablist",
    direction: orientation === "horizontal" ? "row" : "column",
    "aria-orientation": orientation,
    spacing: variant === "pills" ? "md" : 0,
    position,
    grow
  }, panes)), content && /* @__PURE__ */ React__default.createElement("div", {
    role: "tabpanel",
    className: classes.body,
    style: _styles.body
  }, content));
}
Tabs.displayName = "@mantine/core/Tabs";

exports.Tab = Tab.Tab;
exports.Tabs = Tabs;
