// @mantine/core/cjs/components/ColorPicker/Saturation/Saturation.styles.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Thumb_styles = require('../Thumb/Thumb.styles.js');
var createMemoStyles = require('../../../theme/create-memo-styles.js');
var getSizeValue = require('../../../theme/utils/get-size-value/get-size-value.js');

const SATURATION_HEIGHTS = {
  xs: 100,
  sm: 110,
  md: 120,
  lg: 140,
  xl: 160
};
var useStyles = createMemoStyles.createMemoStyles({
  saturationThumb: {},
  saturation: ({ theme, size }) => ({
    boxSizing: "border-box",
    position: "relative",
    height: getSizeValue.getSizeValue({ size, sizes: SATURATION_HEIGHTS }),
    borderRadius: theme.radius.sm,
    margin: getSizeValue.getSizeValue({ size, sizes: Thumb_styles.THUMB_SIZES }) / 2,
    WebkitTapHighlightColor: "transparent",
    "&:focus $saturationThumb": {
      outline: "none",
      boxShadow: `0 0 0 1px ${theme.colorScheme === "dark" ? theme.colors.dark[9] : theme.white}, 0 0 0 3px ${theme.colors[theme.primaryColor][theme.colorScheme === "dark" ? 7 : 5]}`
    },
    "&:focus:not(:focus-visible) $saturationThumb": {
      boxShadow: "none"
    }
  }),
  saturationOverlay: ({ theme, size }) => ({
    position: "absolute",
    boxSizing: "border-box",
    borderRadius: theme.radius.sm,
    top: -getSizeValue.getSizeValue({ size, sizes: Thumb_styles.THUMB_SIZES }) / 2 - 1,
    left: -getSizeValue.getSizeValue({ size, sizes: Thumb_styles.THUMB_SIZES }) / 2 - 1,
    right: -getSizeValue.getSizeValue({ size, sizes: Thumb_styles.THUMB_SIZES }) / 2 - 1,
    bottom: -getSizeValue.getSizeValue({ size, sizes: Thumb_styles.THUMB_SIZES }) / 2 - 1
  })
});

exports.default = useStyles;
