// @mantine/core/esm/components/Menu/MenuBody/MenuBody.styles.js
import { createMemoStyles } from '../../../theme/create-memo-styles.js';
import { getSizeValue } from '../../../theme/utils/get-size-value/get-size-value.js';

const sizes = {
  xs: 120,
  sm: 160,
  md: 200,
  lg: 240,
  xl: 300
};
const MENU_PADDING = 4;
var useStyles = createMemoStyles({
  body: ({ theme, size }) => ({
    width: getSizeValue({ size, sizes }),
    overflow: "hidden",
    backgroundColor: theme.colorScheme === "dark" ? theme.colors.dark[6] : theme.white,
    pointerEvents: "all",
    border: `1px solid ${theme.colorScheme === "dark" ? theme.colors.dark[6] : theme.colors.gray[2]}`,
    padding: MENU_PADDING
  }),
  divider: ({ theme }) => ({
    borderTopColor: theme.colorScheme === "dark" ? theme.colors.dark[7] : theme.colors.gray[2],
    margin: [theme.spacing.xs / 2, -MENU_PADDING]
  }),
  label: ({ theme }) => ({
    color: theme.colorScheme === "dark" ? theme.colors.dark[2] : theme.colors.gray[6],
    fontWeight: 500,
    fontSize: theme.fontSizes.xs,
    padding: [theme.spacing.xs / 2, theme.spacing.sm],
    cursor: "default"
  })
});

export default useStyles;
export { sizes };
