// @mantine/core/cjs/components/Accordion/AccordionControl/AccordionControl.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var cx = require('clsx');
var hooks = require('@mantine/hooks');
var UnstyledButton = require('../../Button/UnstyledButton/UnstyledButton.js');
var ChevronIcon = require('./ChevronIcon.js');
var AccordionControl_styles = require('./AccordionControl.styles.js');
var useMantineTheme = require('../../../theme/use-mantine-theme/use-mantine-theme.js');
var mergeStyles = require('../../../theme/utils/merge-styles/merge-styles.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e['default'] : e; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var cx__default = /*#__PURE__*/_interopDefaultLegacy(cx);

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
function AccordionControl(_a) {
  var _b = _a, {
    opened,
    onToggle,
    label,
    children,
    className,
    classNames,
    styles,
    themeOverride,
    transitionDuration,
    style,
    id
  } = _b, others = __objRest(_b, [
    "opened",
    "onToggle",
    "label",
    "children",
    "className",
    "classNames",
    "styles",
    "themeOverride",
    "transitionDuration",
    "style",
    "id"
  ]);
  const forceUpdate = hooks.useForceUpdate();
  const theme = useMantineTheme.useMantineTheme(themeOverride);
  const reduceMotion = hooks.useReducedMotion();
  const classes = AccordionControl_styles['default']({ theme, transitionDuration: reduceMotion ? 0 : transitionDuration }, classNames, "accordion");
  const _styles = mergeStyles.mergeStyles(classes, styles);
  const [rect, setRect] = React.useState(null);
  hooks.useWindowEvent("resize", () => forceUpdate());
  return /* @__PURE__ */ React__default.createElement("div", __spreadValues({
    className: cx__default(classes.item, { [classes.itemOpened]: opened }, className),
    style: __spreadValues(__spreadValues(__spreadValues({}, _styles.item), opened ? _styles.itemOpened : null), style)
  }, others), /* @__PURE__ */ React__default.createElement(UnstyledButton.UnstyledButton, {
    className: classes.control,
    style: _styles.control,
    onClick: onToggle,
    type: "button",
    "aria-expanded": opened,
    "aria-controls": `${id}-body`,
    id
  }, /* @__PURE__ */ React__default.createElement("div", {
    className: classes.label,
    style: _styles.label
  }, label), /* @__PURE__ */ React__default.createElement("div", {
    className: classes.icon,
    style: _styles.icon
  }, /* @__PURE__ */ React__default.createElement(ChevronIcon.ChevronIcon, null))), /* @__PURE__ */ React__default.createElement("div", {
    className: classes.content,
    role: "region",
    id: `${id}-body`,
    "aria-labelledby": id,
    style: __spreadProps(__spreadValues({}, _styles.content), {
      height: opened ? !rect ? "auto" : rect.getBoundingClientRect().height : 0
    })
  }, /* @__PURE__ */ React__default.createElement("div", {
    className: classes.contentInner,
    style: __spreadProps(__spreadValues({}, _styles.contentInner), { opacity: opened ? 1 : 0 }),
    ref: (node) => setRect(node)
  }, children)));
}
AccordionControl.displayName = "@mantine/core/AccordionControl";

exports.AccordionControl = AccordionControl;
