// @mantine/core/esm/components/Modal/Modal.js
import React from 'react';
import cx from 'clsx';
import { useId, useReducedMotion, useClickOutside, useFocusTrap, useScrollLock } from '@mantine/hooks';
import { CloseButton } from '../ActionIcon/CloseButton/CloseButton.js';
import { Text } from '../Text/Text.js';
import { Paper } from '../Paper/Paper.js';
import { Overlay } from '../Overlay/Overlay.js';
import { Portal } from '../Portal/Portal.js';
import useStyles, { sizes } from './Modal.styles.js';
import { useMantineTheme } from '../../theme/use-mantine-theme/use-mantine-theme.js';
import { mergeStyles } from '../../theme/utils/merge-styles/merge-styles.js';
import { GroupedTransition } from '../Transition/GroupedTransition.js';

var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
const MODAL_SIZES = sizes;
function MantineModal(_a) {
  var _b = _a, {
    className,
    style,
    opened,
    themeOverride,
    title,
    onClose,
    children,
    hideCloseButton = false,
    overlayOpacity,
    size = "md",
    transitionDuration = 300,
    closeButtonLabel,
    overlayColor,
    zIndex = 1e3,
    overflow = "outside",
    transition = "slide-down",
    padding = "lg",
    shadow = "lg",
    id,
    classNames,
    styles
  } = _b, others = __objRest(_b, [
    "className",
    "style",
    "opened",
    "themeOverride",
    "title",
    "onClose",
    "children",
    "hideCloseButton",
    "overlayOpacity",
    "size",
    "transitionDuration",
    "closeButtonLabel",
    "overlayColor",
    "zIndex",
    "overflow",
    "transition",
    "padding",
    "shadow",
    "id",
    "classNames",
    "styles"
  ]);
  const baseId = useId(id);
  const titleId = `${baseId}-title`;
  const bodyId = `${baseId}-body`;
  const reduceMotion = useReducedMotion();
  const theme = useMantineTheme(themeOverride);
  const classes = useStyles({ size, overflow, theme }, classNames, "modal");
  const _styles = mergeStyles(classes, styles);
  const clickOutsideRef = useClickOutside(onClose);
  const focusTrapRef = useFocusTrap();
  const duration = reduceMotion ? 1 : transitionDuration;
  const _overlayOpacity = typeof overlayOpacity === "number" ? overlayOpacity : theme.colorScheme === "dark" ? 0.85 : 0.75;
  useScrollLock(opened);
  return /* @__PURE__ */ React.createElement(GroupedTransition, {
    mounted: opened,
    transitions: {
      modal: { duration, transition },
      overlay: { duration: duration / 2, transition: "fade", timingFunction: "ease" }
    }
  }, (transitionStyles) => /* @__PURE__ */ React.createElement("div", __spreadValues({
    className: cx(classes.root, className),
    style: __spreadValues(__spreadValues({}, style), _styles.root)
  }, others), /* @__PURE__ */ React.createElement("div", {
    className: classes.inner,
    onKeyDownCapture: (event) => event.nativeEvent.code === "Escape" && onClose(),
    style: __spreadValues({ zIndex: zIndex + 1 }, _styles.inner),
    ref: focusTrapRef
  }, /* @__PURE__ */ React.createElement(Paper, {
    themeOverride,
    className: classes.modal,
    shadow,
    padding,
    role: "dialog",
    "aria-labelledby": titleId,
    "aria-describedby": bodyId,
    "aria-modal": true,
    style: __spreadValues(__spreadValues({}, _styles.modal), transitionStyles.modal),
    elementRef: clickOutsideRef,
    tabIndex: -1
  }, (title || !hideCloseButton) && /* @__PURE__ */ React.createElement("div", {
    className: classes.header,
    style: _styles.header
  }, /* @__PURE__ */ React.createElement(Text, {
    id: titleId,
    className: classes.title,
    style: _styles.title,
    themeOverride
  }, title), !hideCloseButton && /* @__PURE__ */ React.createElement(CloseButton, {
    iconSize: 16,
    onClick: onClose,
    "aria-label": closeButtonLabel,
    themeOverride
  })), /* @__PURE__ */ React.createElement("div", {
    id: bodyId,
    className: classes.body,
    style: _styles.body
  }, children))), /* @__PURE__ */ React.createElement("div", {
    style: transitionStyles.overlay
  }, /* @__PURE__ */ React.createElement(Overlay, {
    color: overlayColor || (theme.colorScheme === "dark" ? theme.colors.dark[9] : theme.black),
    opacity: _overlayOpacity,
    zIndex
  }))));
}
function Modal(props) {
  return /* @__PURE__ */ React.createElement(Portal, {
    zIndex: props.zIndex || 1e3
  }, /* @__PURE__ */ React.createElement(MantineModal, __spreadValues({}, props)));
}
Modal.displayName = "@mantine/core/Modal";

export { MODAL_SIZES, MantineModal, Modal };
