// @mantine/core/cjs/components/Menu/MenuBody/MenuBody.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var cx = require('clsx');
var hooks = require('@mantine/hooks');
var Paper = require('../../Paper/Paper.js');
var Divider = require('../../Divider/Divider.js');
var Text = require('../../Text/Text.js');
var MenuItem = require('../MenuItem/MenuItem.js');
var MenuLabel = require('../MenuLabel/MenuLabel.js');
var MenuButton = require('../MenuButton/MenuButton.js');
var MenuBody_styles = require('./MenuBody.styles.js');
var useMantineTheme = require('../../../theme/use-mantine-theme/use-mantine-theme.js');
var mergeStyles = require('../../../theme/utils/merge-styles/merge-styles.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e['default'] : e; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var cx__default = /*#__PURE__*/_interopDefaultLegacy(cx);

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
function getNextItem(active, items) {
  for (let i = active + 1; i < items.length; i += 1) {
    if (!items[i].props.disabled && items[i].type === MenuItem.MenuItem) {
      return i;
    }
  }
  return active;
}
function findInitialItem(items) {
  for (let i = 0; i < items.length; i += 1) {
    if (!items[i].props.disabled && items[i].type === MenuItem.MenuItem) {
      return i;
    }
  }
  return -1;
}
function getPreviousItem(active, items) {
  for (let i = active - 1; i >= 0; i -= 1) {
    if (!items[i].props.disabled && items[i].type === MenuItem.MenuItem) {
      return i;
    }
  }
  if (!items[active] || items[active].type !== MenuItem.MenuItem) {
    return findInitialItem(items);
  }
  return active;
}
function MenuBody(_a) {
  var _b = _a, {
    className,
    style,
    themeOverride,
    opened,
    onClose,
    children,
    size = "md",
    shadow = "md",
    closeOnItemClick = true,
    transitionDuration = 150,
    classNames,
    styles,
    radius,
    trapFocus = true
  } = _b, others = __objRest(_b, [
    "className",
    "style",
    "themeOverride",
    "opened",
    "onClose",
    "children",
    "size",
    "shadow",
    "closeOnItemClick",
    "transitionDuration",
    "classNames",
    "styles",
    "radius",
    "trapFocus"
  ]);
  const items = React__default.Children.toArray(children).filter((item) => item.type === MenuItem.MenuItem || item.type === Divider.Divider || item.type === MenuLabel.MenuLabel);
  const hoveredTimeout = React.useRef();
  const buttonsRefs = React.useRef({});
  const theme = useMantineTheme.useMantineTheme(themeOverride);
  const classes = MenuBody_styles['default']({ size, theme }, classNames, "menu");
  const _styles = mergeStyles.mergeStyles(classes, styles);
  const [hovered, setHovered] = React.useState(findInitialItem(items));
  const focusTrapRef = hooks.useFocusTrap(trapFocus);
  React.useEffect(() => {
    if (!opened) {
      hoveredTimeout.current = window.setTimeout(() => {
        setHovered(findInitialItem(items));
      }, transitionDuration);
    } else {
      window.clearTimeout(hoveredTimeout.current);
    }
    return () => window.clearTimeout(hoveredTimeout.current);
  }, [opened]);
  const handleKeyDown = (event) => {
    const { code } = event.nativeEvent;
    if (code === "Escape") {
      onClose();
    }
    if (code === "Tab" && trapFocus) {
      event.preventDefault();
    }
    if (code === "ArrowUp") {
      event.preventDefault();
      const prevIndex = getPreviousItem(hovered, items);
      setHovered(prevIndex);
      buttonsRefs.current[prevIndex].focus();
    }
    if (code === "ArrowDown") {
      event.preventDefault();
      const nextIndex = getNextItem(hovered, items);
      setHovered(nextIndex);
      buttonsRefs.current[nextIndex].focus();
    }
  };
  if (items.length === 0) {
    return null;
  }
  const buttons = items.map((item, index) => {
    if (item.type === MenuItem.MenuItem) {
      return /* @__PURE__ */ React__default.createElement(MenuButton.MenuButton, __spreadProps(__spreadValues({}, item.props), {
        key: index,
        hovered: hovered === index,
        onHover: () => setHovered(index),
        radius,
        classNames,
        styles,
        onClick: (event) => {
          if (closeOnItemClick) {
            onClose();
          }
          if (typeof item.props.onClick === "function") {
            item.props.onClick(event);
          }
        },
        elementRef: (node) => {
          buttonsRefs.current[index] = node;
        }
      }));
    }
    if (item.type === MenuLabel.MenuLabel) {
      return /* @__PURE__ */ React__default.createElement(Text.Text, __spreadValues({
        key: index,
        className: classes.label,
        style: _styles.label
      }, item.props));
    }
    if (item.type === Divider.Divider) {
      return /* @__PURE__ */ React__default.createElement(Divider.Divider, {
        variant: "solid",
        className: classes.divider,
        margins: theme.spacing.xs / 2,
        style: _styles.divider,
        key: index
      });
    }
    return null;
  });
  return /* @__PURE__ */ React__default.createElement(Paper.Paper, __spreadValues({
    shadow,
    className: cx__default(classes.body, className),
    style: __spreadValues(__spreadValues({}, style), _styles.body),
    onKeyDownCapture: handleKeyDown,
    role: "menu",
    "aria-orientation": "vertical",
    radius,
    onMouseLeave: () => setHovered(-1)
  }, others), /* @__PURE__ */ React__default.createElement("div", {
    ref: focusTrapRef
  }, buttons));
}
MenuBody.displayName = "@mantine/core/MenuBody";

exports.MenuBody = MenuBody;
