// @mantine/core/esm/components/Progress/Progress.js
import React from 'react';
import cx from 'clsx';
import { useReducedMotion } from '@mantine/hooks';
import useStyles, { sizes } from './Progress.styles.js';
import { useMantineTheme } from '../../theme/use-mantine-theme/use-mantine-theme.js';
import { mergeStyles } from '../../theme/utils/merge-styles/merge-styles.js';
import { getThemeColor } from '../../theme/utils/get-theme-color/get-theme-color.js';

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
const PROGRESS_SIZES = sizes;
function getCumulativeSections(sections) {
  return sections.reduce((acc, section) => {
    acc.sections.push(__spreadProps(__spreadValues({}, section), { accumulated: acc.accumulated }));
    acc.accumulated += section.value;
    return acc;
  }, { accumulated: 0, sections: [] }).sections;
}
function Progress(_a) {
  var _b = _a, {
    className,
    style,
    value,
    color,
    size = "md",
    radius = "sm",
    striped = false,
    themeOverride,
    "aria-label": ariaLabel,
    classNames,
    styles,
    sections
  } = _b, others = __objRest(_b, [
    "className",
    "style",
    "value",
    "color",
    "size",
    "radius",
    "striped",
    "themeOverride",
    "aria-label",
    "classNames",
    "styles",
    "sections"
  ]);
  const theme = useMantineTheme(themeOverride);
  const reduceMotion = useReducedMotion();
  const classes = useStyles({ color, size, radius, striped, reduceMotion, theme }, classNames, "progress");
  const _styles = mergeStyles(classes, styles);
  const segments = Array.isArray(sections) ? getCumulativeSections(sections).map((section, index) => /* @__PURE__ */ React.createElement("div", {
    key: index,
    className: classes.bar,
    style: __spreadProps(__spreadValues({}, _styles.bar), {
      width: `${section.value}%`,
      left: `${section.accumulated}%`,
      backgroundColor: getThemeColor({ theme, color: section.color, shade: 7 })
    })
  })) : null;
  return /* @__PURE__ */ React.createElement("div", __spreadValues({
    className: cx(classes.root, className),
    style: __spreadValues(__spreadValues({}, style), _styles.root)
  }, others), segments || /* @__PURE__ */ React.createElement("div", {
    role: "progressbar",
    "aria-valuemax": 100,
    "aria-valuemin": 0,
    "aria-valuenow": value,
    "aria-label": ariaLabel,
    className: classes.bar,
    style: __spreadProps(__spreadValues({}, _styles.bar), { width: `${value}%` })
  }));
}
Progress.displayName = "@mantine/core/Progress";

export { PROGRESS_SIZES, Progress };
