// @mantine/core/esm/components/Burger/Burger.js
import React from 'react';
import cx from 'clsx';
import { useReducedMotion } from '@mantine/hooks';
import useStyles, { sizes } from './Burger.styles.js';
import { useMantineTheme } from '../../theme/use-mantine-theme/use-mantine-theme.js';
import { mergeStyles } from '../../theme/utils/merge-styles/merge-styles.js';

var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
const BURGER_SIZES = sizes;
function Burger(_a) {
  var _b = _a, {
    className,
    style,
    opened,
    color,
    size = "md",
    themeOverride,
    elementRef,
    classNames,
    styles
  } = _b, others = __objRest(_b, [
    "className",
    "style",
    "opened",
    "color",
    "size",
    "themeOverride",
    "elementRef",
    "classNames",
    "styles"
  ]);
  const theme = useMantineTheme(themeOverride);
  const reduceMotion = useReducedMotion();
  const _color = color || (theme.colorScheme === "dark" ? theme.white : theme.black);
  const classes = useStyles({ color: _color, size, theme, reduceMotion }, classNames, "burger");
  const _styles = mergeStyles(classes, styles);
  return /* @__PURE__ */ React.createElement("button", __spreadValues({
    type: "button",
    className: cx(classes.root, className),
    ref: elementRef,
    style: __spreadValues(__spreadValues({}, style), _styles.root)
  }, others), /* @__PURE__ */ React.createElement("div", {
    className: cx(classes.burger, { [classes.opened]: opened }),
    style: _styles.burger
  }));
}
Burger.displayName = "@mantine/core/Burger";

export { BURGER_SIZES, Burger };
