// @mantine/core/esm/components/Slider/Thumb/Thumb.styles.js
import { sizes } from '../SliderRoot/SliderRoot.styles.js';
import { createMemoStyles } from '../../../theme/create-memo-styles.js';
import { getThemeColor } from '../../../theme/utils/get-theme-color/get-theme-color.js';
import { getSizeValue } from '../../../theme/utils/get-size-value/get-size-value.js';
import { getFocusStyles } from '../../../theme/utils/get-focus-styles/get-focus-styles.js';

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var useStyles = createMemoStyles({
  label: ({ theme, color }) => ({
    position: "absolute",
    top: -32,
    backgroundColor: getThemeColor({ theme, color, shade: 6 }),
    fontSize: theme.fontSizes.xs,
    color: theme.white,
    padding: theme.spacing.xs / 2,
    borderRadius: theme.radius.sm,
    whiteSpace: "nowrap"
  }),
  thumb: ({ theme, color, size }) => __spreadProps(__spreadValues({}, getFocusStyles(theme)), {
    boxSizing: "border-box",
    position: "absolute",
    height: getSizeValue({ sizes, size }) * 2,
    width: getSizeValue({ sizes, size }) * 2,
    backgroundColor: theme.colorScheme === "dark" ? getThemeColor({ theme, color, shade: 6 }) : theme.white,
    border: `4px solid ${theme.colorScheme === "dark" ? theme.white : getThemeColor({ theme, color, shade: 6 })}`,
    transform: "translate(-50%, -50%)",
    top: "50%",
    cursor: "pointer",
    borderRadius: 1e3,
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
    transitionDuration: "100ms",
    transitionProperty: "box-shadow, transform",
    transitionTimingFunction: theme.transitionTimingFunction,
    zIndex: 2
  }),
  dragging: ({ theme }) => ({
    transform: "translate(-50%, -50%) scale(1.1)",
    boxShadow: theme.shadows.md
  })
});

export default useStyles;
