// @mantine/core/cjs/components/Popover/PopoverBody/PopoverBody.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var Text = require('../../Text/Text.js');
var CloseButton = require('../../ActionIcon/CloseButton/CloseButton.js');
var PopoverBody_styles = require('./PopoverBody.styles.js');
var useMantineTheme = require('../../../theme/use-mantine-theme/use-mantine-theme.js');
var mergeStyles = require('../../../theme/utils/merge-styles/merge-styles.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e['default'] : e; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
function PopoverBody(_a) {
  var _b = _a, {
    classNames,
    styles,
    themeOverride,
    shadow,
    spacing,
    radius,
    elementRef,
    withCloseButton,
    title,
    titleId,
    bodyId,
    onClose,
    closeButtonLabel,
    children
  } = _b, others = __objRest(_b, [
    "classNames",
    "styles",
    "themeOverride",
    "shadow",
    "spacing",
    "radius",
    "elementRef",
    "withCloseButton",
    "title",
    "titleId",
    "bodyId",
    "onClose",
    "closeButtonLabel",
    "children"
  ]);
  const theme = useMantineTheme.useMantineTheme(themeOverride);
  const classes = PopoverBody_styles['default']({ theme, shadow, radius, spacing }, classNames, "popover");
  const _styles = mergeStyles.mergeStyles(classes, styles);
  return /* @__PURE__ */ React__default.createElement("div", __spreadValues({
    role: "dialog",
    tabIndex: -1,
    "aria-labelledby": titleId,
    "aria-describedby": bodyId,
    className: classes.wrapper,
    style: _styles.wrapper,
    ref: elementRef
  }, others), /* @__PURE__ */ React__default.createElement("div", {
    className: classes.popover,
    style: _styles.popover
  }, /* @__PURE__ */ React__default.createElement("div", {
    className: classes.body,
    style: _styles.body
  }, !!title && /* @__PURE__ */ React__default.createElement("div", {
    className: classes.header,
    style: _styles.header
  }, /* @__PURE__ */ React__default.createElement(Text.Text, {
    size: "sm",
    id: titleId,
    className: classes.title,
    style: _styles.title
  }, title)), withCloseButton && /* @__PURE__ */ React__default.createElement(CloseButton.CloseButton, {
    themeOverride,
    size: "sm",
    onClick: onClose,
    "aria-label": closeButtonLabel,
    className: classes.close,
    style: _styles.close
  }), /* @__PURE__ */ React__default.createElement("div", {
    className: classes.inner,
    id: bodyId,
    style: _styles.inner
  }, children))));
}
PopoverBody.displayName = "@mantine/core/PopoverBody";

exports.PopoverBody = PopoverBody;
