// @mantine/core/cjs/components/Notification/Notification.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var cx = require('clsx');
var Paper = require('../Paper/Paper.js');
var Text = require('../Text/Text.js');
var Loader = require('../Loader/Loader.js');
var CloseButton = require('../ActionIcon/CloseButton/CloseButton.js');
var Notification_styles = require('./Notification.styles.js');
var useMantineTheme = require('../../theme/use-mantine-theme/use-mantine-theme.js');
var mergeStyles = require('../../theme/utils/merge-styles/merge-styles.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e['default'] : e; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var cx__default = /*#__PURE__*/_interopDefaultLegacy(cx);

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
function Notification(_a) {
  var _b = _a, {
    className,
    style,
    color = "blue",
    loading = false,
    disallowClose = false,
    title,
    icon,
    children,
    onClose,
    closeButtonProps,
    themeOverride,
    classNames,
    styles
  } = _b, others = __objRest(_b, [
    "className",
    "style",
    "color",
    "loading",
    "disallowClose",
    "title",
    "icon",
    "children",
    "onClose",
    "closeButtonProps",
    "themeOverride",
    "classNames",
    "styles"
  ]);
  const theme = useMantineTheme.useMantineTheme(themeOverride);
  const classes = Notification_styles['default']({ color, disallowClose, theme }, classNames, "notification");
  const _styles = mergeStyles.mergeStyles(classes, styles);
  const withIcon = icon || loading;
  return /* @__PURE__ */ React__default.createElement(Paper.Paper, __spreadValues({
    shadow: "lg",
    padding: "sm",
    className: cx__default(classes.root, { [classes.withIcon]: withIcon }, className),
    role: "alert",
    themeOverride,
    style: __spreadValues(__spreadValues(__spreadValues({}, style), _styles.root), withIcon ? _styles.withIcon : null)
  }, others), icon && !loading && /* @__PURE__ */ React__default.createElement("div", {
    className: classes.icon,
    style: _styles.icon
  }, icon), loading && /* @__PURE__ */ React__default.createElement(Loader.Loader, {
    size: 28,
    color,
    className: classes.loader,
    style: _styles.loader
  }), /* @__PURE__ */ React__default.createElement("div", {
    className: classes.body,
    style: _styles.body
  }, title && /* @__PURE__ */ React__default.createElement(Text.Text, {
    className: classes.title,
    size: "sm",
    weight: 500,
    themeOverride,
    style: _styles.title
  }, title), /* @__PURE__ */ React__default.createElement(Text.Text, {
    className: classes.description,
    size: "sm",
    themeOverride,
    style: _styles.description
  }, children)), !disallowClose && /* @__PURE__ */ React__default.createElement(CloseButton.CloseButton, __spreadProps(__spreadValues({}, closeButtonProps), {
    iconSize: 16,
    color: "gray",
    onClick: onClose,
    themeOverride
  })));
}
Notification.displayName = "@mantine/core/Notification";

exports.Notification = Notification;
