// @mantine/core/cjs/components/Slider/Track/Track.styles.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var SliderRoot_styles = require('../SliderRoot/SliderRoot.styles.js');
var createMemoStyles = require('../../../theme/create-memo-styles.js');
var getSizeValue = require('../../../theme/utils/get-size-value/get-size-value.js');
var getThemeColor = require('../../../theme/utils/get-theme-color/get-theme-color.js');

var useStyles = createMemoStyles.createMemoStyles({
  track: ({ theme, radius, size }) => ({
    position: "relative",
    height: getSizeValue.getSizeValue({ sizes: SliderRoot_styles.sizes, size }),
    width: "100%",
    backgroundColor: theme.colorScheme === "dark" ? theme.colors.dark[4] : theme.colors.gray[2],
    borderRadius: getSizeValue.getSizeValue({ size: radius, sizes: theme.radius })
  }),
  bar: ({ theme, color, radius }) => ({
    position: "absolute",
    top: 0,
    bottom: 0,
    left: 0,
    backgroundColor: getThemeColor.getThemeColor({ theme, color, shade: 6 }),
    borderRadius: getSizeValue.getSizeValue({ size: radius, sizes: theme.radius })
  })
});

exports.default = useStyles;
