// @mantine/core/cjs/components/ColorPicker/ColorSlider/ColorSlider.styles.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Thumb_styles = require('../Thumb/Thumb.styles.js');
var createMemoStyles = require('../../../theme/create-memo-styles.js');
var getSizeValue = require('../../../theme/utils/get-size-value/get-size-value.js');

var useStyles = createMemoStyles.createMemoStyles({
  sliderThumb: {},
  slider: ({ theme, size }) => ({
    position: "relative",
    height: getSizeValue.getSizeValue({ size, sizes: Thumb_styles.THUMB_SIZES }) + 2,
    boxSizing: "border-box",
    marginLeft: getSizeValue.getSizeValue({ size, sizes: Thumb_styles.THUMB_SIZES }) / 2,
    marginRight: getSizeValue.getSizeValue({ size, sizes: Thumb_styles.THUMB_SIZES }) / 2,
    outline: 0,
    "&:focus $sliderThumb": {
      outline: "none",
      boxShadow: `0 0 0 1px ${theme.colorScheme === "dark" ? theme.colors.dark[9] : theme.white}, 0 0 0 3px ${theme.colors[theme.primaryColor][theme.colorScheme === "dark" ? 7 : 5]}`
    },
    "&:focus:not(:focus-visible) $sliderThumb": {
      boxShadow: "none"
    }
  }),
  sliderOverlay: ({ size }) => ({
    position: "absolute",
    boxSizing: "border-box",
    top: 0,
    bottom: 0,
    left: -getSizeValue.getSizeValue({ size, sizes: Thumb_styles.THUMB_SIZES }) / 2 - 1,
    right: -getSizeValue.getSizeValue({ size, sizes: Thumb_styles.THUMB_SIZES }) / 2 - 1,
    borderRadius: 1e3
  })
});

exports.default = useStyles;
