// @mantine/core/cjs/components/Select/SelectItems/SelectItems.styles.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var createMemoStyles = require('../../../theme/create-memo-styles.js');
var getSizeValue = require('../../../theme/utils/get-size-value/get-size-value.js');

var useStyles = createMemoStyles.createMemoStyles({
  item: ({ theme, size }) => ({
    boxSizing: "border-box",
    textAlign: "left",
    width: "100%",
    padding: [
      getSizeValue.getSizeValue({ size, sizes: theme.spacing }) / 1.5,
      getSizeValue.getSizeValue({ size, sizes: theme.spacing })
    ],
    cursor: "pointer",
    fontSize: getSizeValue.getSizeValue({ size, sizes: theme.fontSizes }),
    color: theme.colorScheme === "dark" ? theme.colors.dark[0] : theme.black,
    borderRadius: theme.radius.sm
  }),
  selected: ({ theme }) => ({
    backgroundColor: theme.colorScheme === "dark" ? theme.colors.dark[7] : theme.colors[theme.primaryColor][0],
    color: theme.colorScheme === "dark" ? theme.white : theme.colors[theme.primaryColor][9]
  }),
  hovered: ({ theme }) => ({
    backgroundColor: theme.colorScheme === "dark" ? theme.colors.dark[4] : theme.colors.gray[1]
  }),
  nothingFound: ({ theme, size }) => ({
    boxSizing: "border-box",
    color: theme.colors.gray[6],
    paddingTop: getSizeValue.getSizeValue({ size, sizes: theme.spacing }) / 2,
    paddingBottom: getSizeValue.getSizeValue({ size, sizes: theme.spacing }) / 2,
    textAlign: "center"
  })
});

exports.default = useStyles;
