// @mantine/core/cjs/components/Select/SelectItems/SelectItems.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var cx = require('clsx');
var Text = require('../../Text/Text.js');
var SelectItems_styles = require('./SelectItems.styles.js');
var useMantineTheme = require('../../../theme/use-mantine-theme/use-mantine-theme.js');
var mergeStyles = require('../../../theme/utils/merge-styles/merge-styles.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e['default'] : e; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var cx__default = /*#__PURE__*/_interopDefaultLegacy(cx);

var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
function SelectItems({
  data,
  hovered,
  themeOverride,
  classNames,
  styles,
  isItemSelected,
  uuid,
  __staticSelector,
  onItemHover,
  onItemSelect,
  itemsRefs,
  itemComponent: Item,
  size,
  nothingFound
}) {
  const theme = useMantineTheme.useMantineTheme(themeOverride);
  const classes = SelectItems_styles['default']({ theme, size }, classNames, __staticSelector);
  const _styles = mergeStyles.mergeStyles(classes, styles);
  const items = data.map((item, index) => {
    const selected = typeof isItemSelected === "function" ? isItemSelected(item.value) : false;
    return /* @__PURE__ */ React__default.createElement(Item, __spreadValues({
      key: item.value,
      className: cx__default(classes.item, {
        [classes.hovered]: hovered === index,
        [classes.selected]: selected
      }),
      style: __spreadValues(__spreadValues(__spreadValues({}, _styles.item), hovered === index ? _styles.hovered : null), selected ? _styles.selected : null),
      onMouseEnter: () => onItemHover(index),
      id: `${uuid}-${index}`,
      role: "option",
      tabIndex: -1,
      "aria-selected": hovered === index,
      elementRef: (node) => {
        if (itemsRefs && itemsRefs.current) {
          itemsRefs.current[item.value] = node;
        }
      },
      onMouseDown: (event) => {
        event.preventDefault();
        onItemSelect(item);
      }
    }, item));
  });
  return items.length > 0 ? /* @__PURE__ */ React__default.createElement(React__default.Fragment, null, items) : /* @__PURE__ */ React__default.createElement(Text.Text, {
    size,
    className: classes.nothingFound,
    style: _styles.nothingFound
  }, nothingFound);
}
SelectItems.displayName = "@mantine/core/SelectItems";

exports.SelectItems = SelectItems;
