// @mantine/core/esm/components/Popover/Popover.js
import React, { useState } from 'react';
import cx from 'clsx';
import { useFocusTrap, useClickOutside, useId, useReducedMotion, useMergedRef } from '@mantine/hooks';
import { Popper } from '../Popper/Popper.js';
import { PopoverBody } from './PopoverBody/PopoverBody.js';
import useStyles from './Popover.styles.js';
import { useMantineTheme } from '../../theme/use-mantine-theme/use-mantine-theme.js';
import { mergeStyles } from '../../theme/utils/merge-styles/merge-styles.js';

var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
function Popover(_a) {
  var _b = _a, {
    className,
    style,
    themeOverride,
    children,
    target,
    title,
    onClose,
    opened,
    zIndex = 1,
    arrowSize = 4,
    withArrow = false,
    transition = "fade",
    transitionDuration = 200,
    transitionTimingFunction,
    gutter = 10,
    position = "left",
    placement = "center",
    disabled = false,
    noClickOutside = false,
    noFocusTrap = false,
    noEscape = false,
    withCloseButton = false,
    radius = "sm",
    spacing = "md",
    shadow = "sm",
    closeButtonLabel,
    positionDependencies = [],
    id,
    classNames,
    styles
  } = _b, others = __objRest(_b, [
    "className",
    "style",
    "themeOverride",
    "children",
    "target",
    "title",
    "onClose",
    "opened",
    "zIndex",
    "arrowSize",
    "withArrow",
    "transition",
    "transitionDuration",
    "transitionTimingFunction",
    "gutter",
    "position",
    "placement",
    "disabled",
    "noClickOutside",
    "noFocusTrap",
    "noEscape",
    "withCloseButton",
    "radius",
    "spacing",
    "shadow",
    "closeButtonLabel",
    "positionDependencies",
    "id",
    "classNames",
    "styles"
  ]);
  const theme = useMantineTheme(themeOverride);
  const classes = useStyles({ theme }, classNames, "popover");
  const _styles = mergeStyles(classes, styles);
  const handleClose = () => typeof onClose === "function" && onClose();
  const [referenceElement, setReferenceElement] = useState(null);
  const [rootElement, setRootElement] = useState(null);
  const [dropdownElement, setDropdownElement] = useState(null);
  const focusTrapRef = useFocusTrap(!noFocusTrap);
  useClickOutside(() => !noClickOutside && handleClose(), null, [rootElement, dropdownElement]);
  const handleKeydown = (event) => {
    if (!noEscape && event.nativeEvent.code === "Escape") {
      handleClose();
    }
  };
  const uuid = useId(id);
  const titleId = `${uuid}-title`;
  const bodyId = `${uuid}-body`;
  return /* @__PURE__ */ React.createElement("div", __spreadValues({
    className: cx(classes.root, className),
    id,
    style: __spreadValues(__spreadValues({}, style), _styles.root),
    ref: setRootElement
  }, others), /* @__PURE__ */ React.createElement(Popper, {
    referenceElement,
    transitionDuration: useReducedMotion() ? 0 : transitionDuration,
    transition,
    mounted: opened && !disabled,
    position,
    placement,
    gutter,
    withArrow,
    arrowSize,
    zIndex,
    arrowClassName: classes.arrow,
    arrowStyle: _styles.arrow,
    forceUpdateDependencies: [radius, shadow, spacing, ...positionDependencies]
  }, /* @__PURE__ */ React.createElement(PopoverBody, {
    shadow,
    radius,
    spacing,
    withCloseButton,
    titleId,
    bodyId,
    closeButtonLabel,
    onClose: handleClose,
    elementRef: useMergedRef(focusTrapRef, setDropdownElement),
    onKeyDownCapture: handleKeydown,
    classNames,
    styles
  }, children)), /* @__PURE__ */ React.createElement("div", {
    className: classes.target,
    style: _styles.target,
    ref: setReferenceElement
  }, target));
}
Popover.displayName = "@mantine/core/Popover";

export { Popover };
