// @mantine/core/esm/components/Checkbox/Checkbox.js
import React from 'react';
import cx from 'clsx';
import { useId, useReducedMotion } from '@mantine/hooks';
import { CheckboxIcon } from './CheckboxIcon.js';
import useStyles, { sizes } from './Checkbox.styles.js';
import { useMantineTheme } from '../../theme/use-mantine-theme/use-mantine-theme.js';
import { mergeStyles } from '../../theme/utils/merge-styles/merge-styles.js';

var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
const CHECKBOX_SIZES = sizes;
function Checkbox(_a) {
  var _b = _a, {
    className,
    style,
    checked,
    onChange,
    color,
    themeOverride,
    label,
    disabled,
    indeterminate,
    id,
    size = "sm",
    wrapperProps,
    elementRef,
    children,
    classNames,
    styles,
    transitionDuration = 100
  } = _b, others = __objRest(_b, [
    "className",
    "style",
    "checked",
    "onChange",
    "color",
    "themeOverride",
    "label",
    "disabled",
    "indeterminate",
    "id",
    "size",
    "wrapperProps",
    "elementRef",
    "children",
    "classNames",
    "styles",
    "transitionDuration"
  ]);
  const uuid = useId(id);
  const theme = useMantineTheme(themeOverride);
  const reduceMotion = useReducedMotion();
  const classes = useStyles({ size, color, theme, transitionDuration: reduceMotion ? 0 : transitionDuration }, classNames, "checkbox");
  const _styles = mergeStyles(classes, styles);
  return /* @__PURE__ */ React.createElement("div", __spreadValues({
    className: cx(classes.root, className),
    style: __spreadValues(__spreadValues({}, style), _styles.root)
  }, wrapperProps), /* @__PURE__ */ React.createElement("div", {
    className: classes.inner,
    style: _styles.inner
  }, /* @__PURE__ */ React.createElement("input", __spreadValues({
    id: uuid,
    ref: elementRef,
    type: "checkbox",
    className: classes.input,
    style: _styles.input,
    checked: indeterminate || checked,
    onChange,
    disabled
  }, others)), /* @__PURE__ */ React.createElement(CheckboxIcon, {
    indeterminate,
    className: classes.icon,
    style: _styles.icon
  })), label && /* @__PURE__ */ React.createElement("label", {
    className: classes.label,
    htmlFor: uuid,
    style: _styles.label
  }, label));
}
Checkbox.displayName = "@mantine/core/Checkbox";

export { CHECKBOX_SIZES, Checkbox };
