// @mantine/core/esm/components/ColorPicker/ColorPicker.js
import React, { useRef, useState, useEffect } from 'react';
import cx from 'clsx';
import { useUncontrolled, useDidUpdate } from '@mantine/hooks';
import { ColorSwatch } from '../ColorSwatch/ColorSwatch.js';
import { HueSlider } from './HueSlider/HueSlider.js';
export { HueSlider } from './HueSlider/HueSlider.js';
import { AlphaSlider } from './AlphaSlider/AlphaSlider.js';
export { AlphaSlider } from './AlphaSlider/AlphaSlider.js';
import { Saturation } from './Saturation/Saturation.js';
import { Swatches } from './Swatches/Swatches.js';
import useStyles from './ColorPicker.styles.js';
import { useMantineTheme } from '../../theme/use-mantine-theme/use-mantine-theme.js';
import { mergeStyles } from '../../theme/utils/merge-styles/merge-styles.js';
import { isColorValid, parseColor } from './converters/parsers.js';
import { convertHsvaTo } from './converters/converters.js';
import { getSizeValue } from '../../theme/utils/get-size-value/get-size-value.js';

var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
const SWATCH_SIZES = {
  xs: 26,
  sm: 34,
  md: 42,
  lg: 50,
  xl: 54
};
function ColorPicker(_a) {
  var _b = _a, {
    value,
    defaultValue,
    onChange,
    format,
    swatches,
    swatchesPerRow = 10,
    size = "sm",
    withPicker = true,
    fullWidth = false,
    focusable = true,
    __staticSelector = "color-picker",
    saturationLabel,
    hueLabel,
    alphaLabel,
    themeOverride,
    className,
    style,
    styles,
    classNames
  } = _b, others = __objRest(_b, [
    "value",
    "defaultValue",
    "onChange",
    "format",
    "swatches",
    "swatchesPerRow",
    "size",
    "withPicker",
    "fullWidth",
    "focusable",
    "__staticSelector",
    "saturationLabel",
    "hueLabel",
    "alphaLabel",
    "themeOverride",
    "className",
    "style",
    "styles",
    "classNames"
  ]);
  const theme = useMantineTheme(themeOverride);
  const classes = useStyles({ theme, size, fullWidth }, classNames, __staticSelector);
  const _styles = mergeStyles(classes, styles);
  const formatRef = useRef(format);
  const withAlpha = format === "rgba" || format === "hsla";
  const [shouldSkip, setShouldSkip] = useState(false);
  const [_value, setValue] = useUncontrolled({
    value,
    defaultValue,
    finalValue: "#FFFFFF",
    rule: (val) => isColorValid(val),
    onChange
  });
  const [parsed, setParsed] = useState(parseColor(_value));
  const handleChange = (color) => {
    setShouldSkip(true);
    Promise.resolve().then(() => setShouldSkip(false));
    setParsed((current) => {
      const next = __spreadValues(__spreadValues({}, current), color);
      setValue(convertHsvaTo(formatRef.current, next));
      return next;
    });
  };
  useEffect(() => {
    if (isColorValid(value) && !shouldSkip) {
      setParsed(parseColor(value));
    }
  }, [value]);
  useDidUpdate(() => {
    formatRef.current = format;
    setValue(convertHsvaTo(format, parsed));
  }, [format]);
  return /* @__PURE__ */ React.createElement("div", __spreadValues({
    className: cx(classes.root, className),
    style: __spreadValues(__spreadValues({}, style), _styles.root)
  }, others), withPicker && /* @__PURE__ */ React.createElement(React.Fragment, null, /* @__PURE__ */ React.createElement(Saturation, {
    value: parsed,
    onChange: handleChange,
    color: _value,
    themeOverride,
    styles,
    classNames,
    size,
    focusable,
    saturationLabel,
    __staticSelector
  }), /* @__PURE__ */ React.createElement("div", {
    className: classes.body,
    style: _styles.body
  }, /* @__PURE__ */ React.createElement("div", {
    className: classes.sliders,
    style: _styles.sliders
  }, /* @__PURE__ */ React.createElement(HueSlider, {
    value: parsed.h,
    onChange: (h) => handleChange({ h }),
    size,
    themeOverride,
    styles,
    classNames,
    focusable,
    "aria-label": hueLabel,
    __staticSelector
  }), withAlpha && /* @__PURE__ */ React.createElement(AlphaSlider, {
    value: parsed.a,
    onChange: (a) => handleChange({ a }),
    size,
    color: convertHsvaTo("hex", parsed),
    style: { marginTop: 6 },
    themeOverride,
    styles,
    classNames,
    focusable,
    "aria-label": alphaLabel,
    __staticSelector
  })), withAlpha && /* @__PURE__ */ React.createElement(ColorSwatch, {
    color: _value,
    radius: "sm",
    themeOverride,
    size: getSizeValue({ size, sizes: SWATCH_SIZES }),
    className: classes.preview,
    style: _styles.preview
  }))), Array.isArray(swatches) && /* @__PURE__ */ React.createElement(Swatches, {
    data: swatches,
    onSelect: handleChange,
    style: { marginTop: 5 },
    swatchesPerRow,
    focusable,
    classNames,
    styles,
    __staticSelector
  }));
}
ColorPicker.displayName = "@mantine/core/ColorPicker";

export { ColorPicker };
