// @mantine/core/esm/components/Tabs/TabControl/TabControl.js
import React from 'react';
import cx from 'clsx';
import { useReducedMotion, useMergedRef } from '@mantine/hooks';
import useStyles from './TabControl.styles.js';
import { useMantineTheme } from '../../../theme/use-mantine-theme/use-mantine-theme.js';
import { mergeStyles } from '../../../theme/utils/merge-styles/merge-styles.js';

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
function TabControl(_a) {
  var _b = _a, {
    className,
    style,
    themeOverride,
    active,
    elementRef,
    tabProps,
    color,
    variant = "default",
    classNames,
    styles,
    orientation = "horizontal",
    icon: __
  } = _b, others = __objRest(_b, [
    "className",
    "style",
    "themeOverride",
    "active",
    "elementRef",
    "tabProps",
    "color",
    "variant",
    "classNames",
    "styles",
    "orientation",
    "icon"
  ]);
  const _a2 = tabProps, { label, icon, color: overrideColor, elementRef: _ } = _a2, props = __objRest(_a2, ["label", "icon", "color", "elementRef"]);
  const theme = useMantineTheme(themeOverride);
  const reduceMotion = useReducedMotion();
  const classes = useStyles({ reduceMotion, color: overrideColor || color, theme, orientation }, classNames, "tabs");
  const _styles = mergeStyles(classes, styles);
  return /* @__PURE__ */ React.createElement("button", __spreadProps(__spreadValues(__spreadValues({}, others), props), {
    tabIndex: active ? 0 : -1,
    className: cx(classes.tabControl, classes[variant], { [classes.tabActive]: active }, className),
    type: "button",
    role: "tab",
    "aria-selected": active,
    ref: useMergedRef(elementRef, tabProps.elementRef),
    style: __spreadValues(__spreadValues(__spreadValues(__spreadValues({}, style), _styles.tabControl), _styles[variant]), active ? _styles.tabActive : null)
  }), /* @__PURE__ */ React.createElement("div", {
    className: classes.tabInner,
    style: _styles.tabInner
  }, icon && /* @__PURE__ */ React.createElement("div", {
    className: classes.tabIcon,
    style: _styles.tabIcon
  }, icon), label && /* @__PURE__ */ React.createElement("div", {
    className: classes.tabLabel,
    style: _styles.tabLabel
  }, label)));
}
TabControl.displayName = "@mantine/core/TabControl";

export { TabControl };
