// @mantine/core/cjs/components/Switch/Switch.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var cx = require('clsx');
var hooks = require('@mantine/hooks');
var Switch_styles = require('./Switch.styles.js');
var useMantineTheme = require('../../theme/use-mantine-theme/use-mantine-theme.js');
var mergeStyles = require('../../theme/utils/merge-styles/merge-styles.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e['default'] : e; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var cx__default = /*#__PURE__*/_interopDefaultLegacy(cx);

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
const SWITCH_SIZES = Switch_styles.sizes;
function Switch(_a) {
  var _b = _a, {
    className,
    color,
    label,
    id,
    style,
    size = "sm",
    radius = "xl",
    themeOverride,
    wrapperProps,
    elementRef,
    children,
    classNames,
    styles
  } = _b, others = __objRest(_b, [
    "className",
    "color",
    "label",
    "id",
    "style",
    "size",
    "radius",
    "themeOverride",
    "wrapperProps",
    "elementRef",
    "children",
    "classNames",
    "styles"
  ]);
  const reduceMotion = hooks.useReducedMotion();
  const theme = useMantineTheme.useMantineTheme(themeOverride);
  const classes = Switch_styles['default']({ size, color, radius, reduceMotion, theme }, classNames, "switch");
  const _styles = mergeStyles.mergeStyles(classes, styles);
  const uuid = hooks.useId(id);
  return /* @__PURE__ */ React__default.createElement("div", __spreadValues({
    className: cx__default(classes.root, className),
    style: __spreadValues(__spreadValues({}, style), _styles.root)
  }, wrapperProps), /* @__PURE__ */ React__default.createElement("input", __spreadProps(__spreadValues({}, others), {
    id: uuid,
    ref: elementRef,
    type: "checkbox",
    className: classes.input,
    style: _styles.input
  })), label && /* @__PURE__ */ React__default.createElement("label", {
    className: classes.label,
    htmlFor: uuid,
    style: _styles.label
  }, label));
}
Switch.displayName = "@mantine/core/Switch";

exports.SWITCH_SIZES = SWITCH_SIZES;
exports.Switch = Switch;
