// @mantine/core/cjs/components/Spoiler/Spoiler.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var cx = require('clsx');
var hooks = require('@mantine/hooks');
var Button = require('../Button/Button.js');
var Spoiler_styles = require('./Spoiler.styles.js');
var useMantineTheme = require('../../theme/use-mantine-theme/use-mantine-theme.js');
var mergeStyles = require('../../theme/utils/merge-styles/merge-styles.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e['default'] : e; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var cx__default = /*#__PURE__*/_interopDefaultLegacy(cx);

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
function Spoiler(_a) {
  var _b = _a, {
    className,
    style,
    children,
    maxHeight = 100,
    hideLabel,
    showLabel,
    themeOverride,
    transitionDuration = 200,
    controlRef,
    initialState = false,
    classNames,
    styles
  } = _b, others = __objRest(_b, [
    "className",
    "style",
    "children",
    "maxHeight",
    "hideLabel",
    "showLabel",
    "themeOverride",
    "transitionDuration",
    "controlRef",
    "initialState",
    "classNames",
    "styles"
  ]);
  const classes = Spoiler_styles['default']({
    transitionDuration: !hooks.useReducedMotion() && transitionDuration,
    theme: useMantineTheme.useMantineTheme(themeOverride)
  }, classNames, "spoiler");
  const _styles = mergeStyles.mergeStyles(classes, styles);
  const [show, setShowState] = React.useState(initialState);
  const [spoiler, setSpoilerState] = React.useState(initialState);
  const contentRef = React.useRef(null);
  const spoilerMoreContent = show ? hideLabel : showLabel;
  React.useEffect(() => {
    setSpoilerState(maxHeight < contentRef.current.clientHeight);
  }, [maxHeight, children]);
  return /* @__PURE__ */ React__default.createElement("div", __spreadValues({
    className: cx__default(classes.root, className),
    style: __spreadValues(__spreadValues({}, style), _styles.root)
  }, others), /* @__PURE__ */ React__default.createElement("div", {
    className: classes.content,
    style: __spreadProps(__spreadValues({}, _styles.content), {
      maxHeight: !show ? maxHeight : contentRef.current && contentRef.current.clientHeight
    })
  }, /* @__PURE__ */ React__default.createElement("div", {
    ref: contentRef
  }, children)), spoiler && /* @__PURE__ */ React__default.createElement(Button.Button, {
    variant: "link",
    elementRef: controlRef,
    onClick: () => setShowState((opened) => !opened),
    themeOverride,
    className: classes.control,
    style: _styles.control
  }, spoilerMoreContent));
}
Spoiler.displayName = "@mantine/core/Spoiler";

exports.Spoiler = Spoiler;
